packages/react-components/react-components/src/AccessibilityScenarios/Fields.stories.tsx

```diff
@@ -0,0 +1,521 @@
+                  id="email"
+                  disabled={!isSendNewsletter}
+                  label="E-mail:"
+                  hint="We will send you newsletter to this e-mail."
+                  validationState={errors.email?.types ? 'error' : 'success'}
+                  validationMessage={
+                    !errors.email?.types ? undefined : (
+                      <ValidationMessage id="email" formValidation={formValidation}>
+                        {'required' in errors.email.types ? (
+                          <p>E-mail is required.</p>
+                        ) : (
+                          <>
+                            <p>E-mail is invalid. It must:</p>
+                            <ul>
+                              {'validEmail' in errors.email.types && (
+                                <li>Be a valid e-mail address, like name@example.com.</li>
+                              )}
+                            </ul>
+                          </>
+                        )}
+                      </ValidationMessage>
+                    )
+                  }
+                  aria-required={isSendNewsletter}
+                  aria-invalid={!!errors.email}
+                />
+              }
+              rules={{
+                required: isSendNewsletter,
+                validate: {
+                  validEmail: value => !isSendNewsletter || regexes.validEmail.test(value),
+                  always: () => {
+                    if (!formState.isSubmitting) {
+                      formValidation.onFieldValidated('email');
+                    }
+                    return true;
+                  },
+                },
+              }}
+            />
+
+            <Controller
+              name="acceptTerms"
+              control={control}
+              as={
+                <CheckboxField
+                  id="acceptTerms"
+                  label="I accept terms and conditions"
+                  hint={
+                    <>
+                      Check this field to confirm you have read and understand the <a href="#">terms and conditions</a>.
+                    </>
+                  }
+                  validationState={errors.acceptTerms?.types ? 'error' : 'success'}
+                  validationMessage={
+                    !errors.acceptTerms?.types ? undefined : (
+                      <ValidationMessage id="acceptTerms" formValidation={formValidation}>
+                        {'required' in errors.acceptTerms.types && (
+                          <p>You have to accept the terms and conditions in order to order your ticket.</p>
+                        )}
+                      </ValidationMessage>
+                    )
+                  }
+                  aria-required="true"
+                  aria-invalid={!!errors.acceptTerms}
+                />
+              }
+              rules={{
+                required: true,
+                validate: {
+                  always: () => {
```

Comment:
This form field is always invalid, regardless of the checkbox state

Reply:
I think the problem is in the CheckboxField component, because the same validation is working with the Checkbox component and others. Will send a issue for this  ince it is created by @jurokapsiar .